src/middlewared/middlewared/plugins/system.py

```diff
@@ -632,10 +668,13 @@ async def feature_enabled(self, name):
 
     @private
     async def _system_serial(self):
-        return (await(await Popen(
-            ['dmidecode', '-s', 'system-serial-number'],
-            stdout=subprocess.PIPE,
-        )).communicate())[0].decode().strip() or None
+
+        serial = (await self.middleware.call('system.dmidecode_info'))['system-serial']
```

Comment:
`return (await self.middleware.call('system.dmidecode_info'))['system-serial'] or None`

is an idea